Firestore/core/src/util/executor_std.cc

```diff
@@ -79,58 +109,58 @@ ExecutorStd::~ExecutorStd() {
 }
 
 void ExecutorStd::Execute(Operation&& operation) {
-  PushOnSchedule(std::move(operation), Immediate());
+  PushOnSchedule(Immediate(), kNoTag, std::move(operation));
 }
 
 DelayedOperation ExecutorStd::Schedule(const Milliseconds delay,
-                                       TaggedOperation&& tagged) {
+                                       Tag tag,
+                                       Operation&& operation) {
   // While negative delay can be interpreted as a request for immediate
   // execution, supporting it would provide a hacky way to modify FIFO ordering
   // of immediate operations.
   HARD_ASSERT(delay.count() >= 0, "Schedule: delay cannot be negative");
 
-  namespace chr = std::chrono;
-  const auto now = chr::time_point_cast<Milliseconds>(chr::steady_clock::now());
-  const auto id =
-      PushOnSchedule(std::move(tagged.operation), now + delay, tagged.tag);
+  const auto target_time = MakeTargetTime(delay);
+  const auto id = PushOnSchedule(target_time, tag, std::move(operation));
+  return DelayedOperation(this, id);
+}
 
-  return DelayedOperation{[this, id] { TryCancel(id); }};
+void ExecutorStd::Complete(Task*) {
+  // No-op in this implementation
```

Comment:
There are a few alternatives:

  * Make `Task` take a completion callback instead of an `executor`
  * Make `ExecutorLibdispatch` wrap the operation in another function-like thing (though I avoided this here because it's tricky to propagate cancelation without making operation some type other than `std::function`.

I think the second option isn't terribly viable but I can see a way to making the former work if you prefer it.